packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -117,7 +120,16 @@ class RequestWrapper {
     assert.atLeastOne({request});
 
     const cache = await this.getCache();
-    return await cache.match(request, this.matchOptions);
+    const response = await cache.match(request, this.matchOptions);
+
+    // Give the cacheWillMatch callbacks a chance to modify the cached response.
+    // TODO: Right now, we run all the callbacks sequentially, and return the
+    // final result. Instead, it might make sense to enforce a limit of a single
+    // cacheWillMatch callback, and return an error if there are multiple ones.
+    const callbacks = this.behaviorCallbacks.cacheWillMatch || [];
+    return callbacks.reduce((cachedResponse, callback) => {
+      return callback({cachedResponse});
```

Comment:
Yeah, it is weird. I struggled with this, as mentioned in the "TODO:" comment about whether it made sense to support multiple callbacks.

My takeaway from your comment is that it would be simpler if we just enforced a limit of 1 callback (similar to what we do for `cacheWillUpdate`), and that would end up being much less of a surprise to the developer who accidentally supplies multiple callbacks and doesn't understand how they all interact with each other.